internal/services/signalr/web_pubsub_private_linked_service_data_source.go

```diff
@@ -0,0 +1,95 @@
+package signalr
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/signalr/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+func dataSourceWebPubsubPrivateLinkResource() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceWebPubsubPrivateLinkResourceRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"web_pubsub_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.WebPubsubID,
+			},
+
+			"shared_private_link_resource_types": {
+				Type:     pluginsdk.TypeSet,
```

Comment:
a Data Source cannot contain a Set - this needs to be a List

furthermore: what's the intention of this data source? why isn't this part of the WebPubSub data source?

Reply:
This data source is intended to provide a reference of supported sub resource that can be linked by a web pubsub resource, just as the same as the purpose of `azurerm_monitor_diagnostic_categories`. I thought it would be more straight-forward to get this information separately since user may would only like to refer to such information?

Reply:
@tombuildsstuff may I know if you have any other concerns that I can help tp address? :)

Reply:
In what instances would a user need this information? I searched the repo and I cannot find any other services with such a private link resource data source?

Reply:
My first thought was to provide a reference to the user, just in case they don't know what information to put in the property`subresource_name`. I can remove it if it looks like unnecessary.

Reply:
that is the entire resource thou? thjats why i am asking what is the use to a user for this resoruce?

Reply:
sorry I don't quite follow your concerns. Let me explain future:
the supported subname varies from different resources, for example, the output of this datasource is:
```
webpubsub_private_supported_resource = {
  "id" = "/subscriptions/733b9054-5ed8-4f10-a444-c32458359d40/resourceGroups/xiaxintest/providers/Microsoft.SignalRService/WebPubSub/acctestWebPubsub"
  "shared_private_link_resource_types" = tolist([
    {
      "description" = "Azure App Service can be used as an upstream"
      "subresource_name" = "sites"
    },
    {
      "description" = "Azure Key Vault can be used as credentials store"
      "subresource_name" = "vault"
    },
  ])

```

The user can put either site or vault in the property `subname` in below tf block:
```
resource "azurerm_web_pubsub_shared_private_link_resource" "test" {
  name               = "acctest-%d"
  web_pubsub_id      = azurerm_web_pubsub.test.id
  subresource_name   = "vault"
  target_resource_id = azurerm_key_vault.test.id
}
```